src/Nethermind/Nethermind.Trie/NibbleExtensions.cs

```diff
@@ -84,5 +84,33 @@ public static byte ToByte(Nibble highNibble, Nibble lowNibble)
         {
             return (byte)(((byte)highNibble << 4) | (byte)lowNibble);
         }
+
+        public static byte[] ToBytes(byte[] nibbles)
+        {
+            byte[] bytes = new byte[nibbles.Length / 2];
```

Comment:
May need to ceil it.

Reply:
Or assert that nibble.Length % 2 == 0.

Reply:
@flcl42 could you address that?

Reply:
It needs refactoring, for now we can indeed just add an assertion